### PR TITLE
Make sure evaluation does not hang in detached iframes

### LINKED ISSUE #871: Make sure evaluation does not hang in detached iframes
https://github.com/kblok/puppeteer/pull/15/commits/9083c111caf5663ee534ff5e30127027d424974e

### PR DESCRIPTION
**Pull request by Meir017**
closes #871

### CODE REVIEW

File: lib/PuppeteerSharp/Frame.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
-using Newtonsoft.Json.Linq;
-using PuppeteerSharp.Helpers;
-
+using Newtonsoft.Json.Linq;
+using PuppeteerSharp.Helpers;
```
Comment (kblok) [inline lib/PuppeteerSharp/Frame.cs]: Can we reduce the number of changes on this file?

File: lib/PuppeteerSharp/Frame.cs [hunk 2]
```diff
@@ -257,7 +257,14 @@ public async Task<JSHandle> EvaluateFunctionHandleAsync(string function, params
         /// Gets the <see cref="ExecutionContext"/> associated with the frame.
         /// </summary>
         /// <returns><see cref="ExecutionContext"/> associated with the frame.</returns>
-        public Task<ExecutionContext> GetExecutionContextAsync() => _contextResolveTaskWrapper.Task;
+        public Task<ExecutionContext> GetExecutionContextAsync()
+        {
+            if(Detached)
```
Comment (kblok) [inline lib/PuppeteerSharp/Frame.cs]:
```suggestion
            if (Detached)
```

File: lib/PuppeteerSharp/Frame.cs (hunk 1 above)
Comment (kblok) [reply lib/PuppeteerSharp/Frame.cs]: Hard to see what you changed with so many changes.

File: lib/PuppeteerSharp/Frame.cs [hunk 3]
```diff
@@ -257,7 +257,14 @@ public async Task<JSHandle> EvaluateFunctionHandleAsync(string function, params
         /// Gets the <see cref="ExecutionContext"/> associated with the frame.
         /// </summary>
         /// <returns><see cref="ExecutionContext"/> associated with the frame.</returns>
-        public Task<ExecutionContext> GetExecutionContextAsync() => _contextResolveTaskWrapper.Task;
+        public Task<ExecutionContext> GetExecutionContextAsync()
+        {
+            if (Detached)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Frame.cs]: @kblok this is the only read change in the file
I think the rest are from a Ctrl+K+D in vs...

**Pull request merged**